packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1156,6 +1161,214 @@ class EndpointIndex<TInterval extends ISerializableInterval> implements Interval
 	}
 }
 
+/**
+ * Collection of intervals.
+ *
+ * Provide additional APIs to support efficiently querying a collection of intervals whose endpoints fall within a specified range.
+ */
+export interface IEndpointInRangeIndex<TInterval extends ISerializableInterval>
+	extends IntervalIndex<TInterval> {
+	/**
+	 * @returns an array of all intervals contained in this collection whose endpoints locate in the range [start, end] (includes both ends)
+	 */
+	findIntervalsWithEndpointInRange(start: number, end: number);
+}
+
+/**
+ * Collection of intervals.
+ *
+ * Provide additional APIs to support efficiently querying a collection of intervals whose startpoints fall within a specified range.
+ */
+export interface IStartpointInRangeIndex<TInterval extends ISerializableInterval>
+	extends IntervalIndex<TInterval> {
+	/**
+	 * @returns an array of all intervals contained in this collection whose startpoints locate in the range [start, end] (includes both ends)
+	 */
+	findIntervalsWithStartpointInRange(start: number, end: number);
+}
+
+/**
+ * Interface for intervals that have comparison override properties.
+ */
+const forceCompare = Symbol();
+
+interface HasComparisonOverride {
+	[forceCompare]: number;
+}
+
+/**
+ * Compares two objects based on their comparison override properties.
+ * @returns A number indicating the order of the intervals (-1 for a is lower than b, 0 for tie, 1 for a is greater than b).
+ */
+function compareOverrideables(a: any, b: any): number {
+	const forceCompareA = (a as HasComparisonOverride)[forceCompare] || 0;
+	const forceCompareB = (b as HasComparisonOverride)[forceCompare] || 0;
```

Comment:
```suggestion
function compareOverrideables(a: Partial<HasComparisonOverride>, b: Partial<HasComparisonOverride>): number {
	const forceCompareA = a[forceCompare] ?? 0;
	const forceCompareB = b[forceCompare] ?? 0;
```

- `??` should be mostly equivalent to `||` here, but it tends to have less pitfalls. i'd prefer it whenever possible (don't rely on getting falsy/truthy values in JS to be right)
- always try to type function parameters reasonably :)